### PR TITLE
make _bufferEmpty more efficient

### PR DESCRIPTION
**Pull request by iAmmarTahir**
This PR improves the efficiency of _emptyBuffer function by just checking the first element of the array.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -31,25 +31,19 @@ class Remedian {
...
+   */
   _bufferEmpty(position) {
-    const array = this.buffers[position];
-    for (let j = 0; j < array.length; j++) {
-      if (array[j] != undefined) {
-        return false;
-      }
-    }
-    return true;
+    return this.buffers[position][0] != undefined ? false : true;
```
Comment (VMois) [inline index.js]:
```suggestion
    return this.buffers[position][0] == undefined
```

**Review by VMois — COMMENTED** [review]
Hey. Thank you for your contribution! I didn't expect the reaction to be so fast. Are you a bot? :) I have a few suggestions, and after that, I can merge your PR:

- Please, follow the linting specified in the `eslint.config.js` file.

- Add `closes #2` to your commit message so it is easier to link a specific commit to the issue it resolves.

- I would not write `fix:` in the commit message. `make _bufferEmpty more efficient` is good enough. You can add more details in the next line.

- check my code comment below.

File: index.js (hunk 1 above)
Comment (iAmmarTahir) [reply index.js]: Yeah, that was quick :). Actually, this is my first Open Source PR that's why there were some issues. I have resolved the things you mentioned.

Comment (VMois) [reply index.js]: No problem if this is your first contribution. There are still some things to fix. I will text you later today with some help on how to fix things.

**Comment by VMois** [comment]
So a few more suggestions. The desired change here is only a few lines of code and one clear commit message. As of now, your changes still have linting issues (you have used your own linting instead of following the one in the repository). Plus, you have 4 commits, which is unnecessary, one is enough. This is how you can fix things so we can merge your PR:

1. On your branch, run `git reset --soft HEAD~4`. It will uncommit the last 4 commits without deleting your changes. You will get uncommitted changes.

2. Run `npx eslint index.js` to fix linting issues. Also, undo your changes in `.eslintrc.js` - they are unnecessary.

3. Now you can make a commit. Your commit message should look like this:

```
make _bufferEmpty more efficient

closes #2
```
(note: "closes #2" is in a new line)

Why add "closes"? As soon as I will merge your PR with "closes" in a commit message, GitHub will close an issue related to that. In addition, it is useful for a commit history.

4. Push your new changes using `git push origin buffer-empty-fix -f`. Note the `-f` option. It will overwrite local commit history (with one clear commit) over your remote history (branch on GitHub). There is nothing bad in using `-f` on your side branches (like this one), but you need to be careful not to lose some work.

To check yourself, go through the difference between `main` and your PR branch. You can easily see the difference here on a GitHub in the "Files changed" tab. 

Hope it helps.

File: index.js [hunk 2]
```diff
@@ -190,4 +184,4 @@ class Remedian {
   }
 }
 
-module.exports = Remedian;
+module.exports = Remedian;
```
Comment (VMois) [inline index.js]: *suggestion:* remove your change here.

File: index.js [hunk 3]
```diff
@@ -6,7 +6,7 @@
  * @author VMois (Vladyslav Moisieienkov)
  */
 
-class Remedian {
+ class Remedian {
```
Comment (VMois) [inline index.js]: *suggestion:* remove your change here.

File: .eslintrc.js [hunk 4]
```diff
@@ -15,4 +15,4 @@ module.exports = {
     'require-jsdoc' : 0,
     'max-len': 1,
   },
-};
+};
```
Comment (VMois) [inline .eslintrc.js]: *suggestion:* remove your change here.

**Comment by iAmmarTahir** [comment]
> So a few more suggestions. The desired change here is only a few lines of code and one clear commit message. As of now, your changes still have linting issues (you have used your own linting instead of following the one in the repository). Plus, you have 4 commits, which is unnecessary, one is enough. This is how you can fix things so we can merge your PR:
> 
> 1. On your branch, run `git reset --soft HEAD~4`. It will uncommit the last 4 commits without deleting your changes. You will get uncommitted changes.
> 2. Run `npx eslint index.js` to fix linting issues. Also, undo your changes in `.eslintrc.js` - they are unnecessary.
> 3. Now you can make a commit. Your commit message should look like this:
> 
> ```
> make _bufferEmpty more efficient
> 
> closes #2
> ```
> 
> (note: "closes #2" is in a new line)
> 
> Why add "closes"? As soon as I will merge your PR with "closes" in a commit message, GitHub will close an issue related to that. In addition, it is useful for a commit history.
> 
> 4. Push your new changes using `git push origin buffer-empty-fix -f`. Note the `-f` option. It will overwrite local commit history (with one clear commit) over your remote history (branch on GitHub). There is nothing bad in using `-f` on your side branches (like this one), but you need to be careful not to lose some work.
> 
> To check yourself, go through the difference between `main` and your PR branch. You can easily see the difference here on a GitHub in the "Files changed" tab.
> 
> Hope it helps.

Yeah, thanks a lot for such detailed explanation. It really helped me.

**Comment by VMois** [comment]
> Yeah, thanks a lot for such detailed explanation. It really helped me.

Glad it helped :)

One last small thing. You still have two commits. Make only one commit. After that, I will merge your PR.

**Pull request closed without merging**